src/lib.rs

```diff
@@ -483,49 +485,37 @@ impl Unicorn {
         }
     }
 
-    /// Write an unsigned value register.
-    ///
-    /// Note : The register is defined as an i32 to be able to support the
-    /// different register types (`RegisterX86`, `RegisterARM`, `RegisterMIPS` etc.).
-    /// You need to cast the register with `as i32`.
-    pub fn reg_write(&self, regid: i32, value: u64) -> Result<(), Error> {
-        let p_value: *const u64 = &value;
+    fn reg_write_generic<T: Sized>(&self, regid: i32, value: T) -> Result<()> {
```

Comment:
Like `reg_read_generic`, i believe this should be `unsafe`.

Reply:
Agree, changed.